[ESQL] Improve `skip_unavailable` description in CCS doc (#1919)
elastic/docs-content: explore-analyze/query-filter/languages/esql-cross-clusters.md

## Optional remote clusters [ccq-skip-unavailable-clusters]
```{applies_to}

> stack: ga 9.1

Reviewer: ```suggestion
stack: preview 9.0, ga 9.1
```
this isn't brand new and the docs now cover all 9.x versions, so the reader might be on 9.0 or 9.1 so we need to maintain lifecycle information

Reviewer: but if the entire page is marked `preview 9.0, GA 9.1`, you might not need this section-level tag at all?

Reviewer: @smalyshev in this case you should use [tabs](https://elastic.github.io/docs-builder/syntax/tabs/) to house the 9.0 and 9.1 differences:

``````markdown

::::{tab-set}

:::{tab-item} 9.0
This is where the content for tab #1 goes.
:::

:::{tab-item} 9.1
This is where the content for tab #2 goes.
:::

::::

Author: Oh interesting. I'll try that.

```

> If the remote cluster is configured with `skip_unavailable: true` (the default setting), the cluster would be set
> to `skipped` or `partial` status but the query will not fail, if:

Reviewer: ```suggestion
Remote clusters are configured with `skip_unavailable: true` by default. With this setting, the cluster will be marked as `skipped` or `partial`:

* `skipped` means the cluster was unavailable and no data was returned
* `partial` means some data was returned despite errors during the query

This means queries won't fail if:

* The remote cluster becomes disconnected from the querying cluster (before or during query execution)
* The remote cluster lacks the requested index  
* An error occurs during query processing on the remote cluster
```

Reviewer: attempt to improve readability

* The remote cluster is disconnected from the querying cluster, either before or during the query.
* The remote cluster does not have the requested index.
* An error happened while processing the query on the remote cluster.

> The `partial` status will be used if the remote query was partially successful and some data was returned.

Reviewer: ```suggestion
```

I suggested explaining skipped and partial in the opening section, so can remove this if accept that :)

This however does not apply to the situation when the remote cluster is missing an index and this is the only index in the query,

> or all the indices in the query are missing. For example, the following queries will fail:

Reviewer: This first sentence is a little hard to read, I wonder if we can boil it town to a simpler message:

_Queries will fail when `skip_unavailable: true`, if none of the specified indices exist._

```esql

> FROM cluster_one:missing-index | LIMIT 10
> FROM cluster_one:missing-index* | LIMIT 10
> FROM cluster_one:missing-index*,cluster_two:missing-index | LIMIT 10
> ```

Reviewer: ``````suggestion
FROM cluster_one:missing-index | LIMIT 10 <1>
FROM cluster_one:missing-index* | LIMIT 10 <2>
FROM cluster_one:missing-index*,cluster_two:missing-index | LIMIT 10 <3>
```

1. Single missing index on one cluster
2. Index pattern that matches nothing
3. All specified indices missing across multiple clusters

``````

Author: I think the previous text already explains it, do we really need to supply that much detail? It's just different examples of how to write missing indices, after all.

> ## Optional remote clusters [ccq-skip-unavailable-clusters]

Reviewer 2: Skipping problematic remote clusters?
Fault tolerance for remote clusters?

> 

Author: It's not only unavailable clusters, at least for 9.1 it covers a lot more errors. It is linked to how `skip_unavailable` option behaves, but despite it's name, it controls more than just mere un-availablity now. That's why we need the whole 9.1 section to explain that :)

> 

Reviewer: @smalyshev fair enough, but my main point is to add a sentence to introduce the tab set for the reader, and I trust you to get the details right :)

::::{tab-set}

:::{tab-item} 9.1
Remote clusters are configured with `skip_unavailable: true` setting by default. With this setting, the cluster would be
set
to `skipped` or `partial` status but the query will not fail, if:

* The remote cluster is disconnected from the querying cluster, either before or during the query execution.
* The remote cluster does not have the requested index, or it is not accessible due to security settings.
* An error happened while processing the query on the remote cluster.

> The `partial` status will be used if the remote query was partially successful and some data may have been returned.

Reviewer: thanks @smalyshev this is a nice detail, so I updated my suggestion
